### PR TITLE
Use consistent binding logic for nested properties in object argument values

### LINKED ISSUE #417: Not able to specify complex parameter
Hi,

I am trying to specify a complex parameter for subject and body for Serilog.Sinks.Email like this:

```
    "WriteTo": [
      {
        "Name": "Email",
        "Args": {
          "options": {
            "subject": {
              "type": "Serilog.Formatting.Display.MessageTemplateTextFormatter, Serilog",
              "outputTemplate": "Serilog test"
            },
            "from": "from@email.com",
            "to": [ "to@email.com" ],
            "host": "localhost",
            "body": {
              "type": "Serilog.Formatting.Display.MessageTemplateTextFormatter, Serilog",
              "outputTemplate": "[{Timestamp:HH:mm:ss} {Level}] {SourceContext}{NewLine}{Message}{NewLine}{Exception}{NewLine}"
            }
          },
          "batchingOptions": {
            "batchSizeLimit": 10,
            "period": "00:00:01"
          },
          "restrictedToMinimumLevel": "Information"
        }
      }
    ]
```

Unfortunately it doesn't work for subject and body. It works for the other settings.
If I add a constructor to the options class, that mirrors the properties, it will work.

Is this expected behavior, or is it a bug?
In other words, should this be fixed in Serilog.Settings.Configuration, or in Serilog.Sinks.Email?

Best regards

### PR DESCRIPTION
**Pull request by nblumhardt**
Fixes #417 

When an argument value is a complex object, instead of deferring to `Microsoft.Extensions.Configuration` we'll now recursively use the same logic to construct its properties that we do when constructing the object itself. This means that values accepted for constructor parameters and method arguments now line up with those accepted for property setters.

This should allow for some simplification of the rest of `ObjectArgumentValue` once we've shaken out any issues.

Thanks for your patience and effort narrowing this down, @MrPsi - sorry it took a while to get back around to it! I think the limitation here is the one to fix, so if/when this is merged I think it will supersede https://github.com/serilog/serilog-sinks-email/pull/131.

### CODE REVIEW

File: test/Serilog.Settings.Configuration.Tests/ObjectArgumentValueTests.cs [hunk 1]
```diff
@@ -1259,19 +1260,54 @@ public void ConvertToPlainStruct()
...
+    // This is intended to mirror Serilog.Sinks.Email's options type.
+    // https://github.com/serilog/serilog-settings-configuration/issues/417
+    public class NestedComplexType
+    {
+        public string? Host { get; set; }
+        public ITextFormatter? Subject { get; set; }
+    }
+
+    [Fact]
+    public void ConstructsNestedComplexObjects()
```
Comment (nblumhardt) [inline test/Serilog.Settings.Configuration.Tests/ObjectArgumentValueTests.cs:1288]: Relevant test case is this one :-)

**Comment by MrPsi** [comment]
Thanks @nblumhardt, 

I have tested your PR on my case and it seems to work great. Hopefully it will be included in the next release :)

**Comment by nblumhardt** [comment]
Great! Thanks for the feedback @MrPsi. I'll leave this open a few more days in case there's some feedback to come, but should get a dev build out soon.

**Comment by ArieGato** [comment]
Hi @nblumhardt, do you have any indicatation when the next release will be containing this merge?

**Comment by nblumhardt** [comment]
8.0.3 is out now; thanks @ArieGato 👍

**Pull request merged**